src/iocore/cache/CacheVC.cc

```diff
@@ -535,6 +519,40 @@ LramHit: {
   return EVENT_RETURN; // allow the caller to release the volume lock
 }
 
+bool
+CacheVC::check_ram_cache()
+{
+  int64_t o           = dir_offset(&this->dir);
+  int ram_hit_state   = this->stripe->ram_cache->get(read_key, &this->buf, static_cast<uint64_t>(o));
+  f.compressed_in_ram = (ram_hit_state > RAM_HIT_COMPRESS_NONE) ? 1 : 0;
+  return ram_hit_state >= RAM_HIT_COMPRESS_NONE;
+}
+
+bool
+CacheVC::check_last_open_read_call()
+{
+  if (*this->read_key == this->stripe->first_fragment_key && dir_offset(&this->dir) == this->stripe->first_fragment_offset) {
+    this->buf = this->stripe->first_fragment_data;
+    return true;
+  }
+  return false;
+}
+
+bool
+CacheVC::check_aggregation_buffer()
```

Comment:
I'm not good at naming, but this function does more than check, allocate new buffer and copy. Any other idea?

Reply:
Maybe all those methods could be named something like `load_from_aggregation_buffer`. What do you think of that name? Do you have any ideas?

Reply:
`load_from_aggregation_buffer` makes sense 👍